src/panel/category/CategoryPanel.jsx

```diff
@@ -0,0 +1,106 @@
+/* global _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Panel from 'src/components/ui/Panel';
+import PoiCategoryItemList from './PoiCategoryItemList';
+import CategoryPanelError from './CategoryPanelError';
+import CategoryPanelHeader from './CategoryPanelHeader';
+import layouts from 'src/panel/layouts.js';
+import Telemetry from 'src/libs/telemetry';
+
+export default class CategoryPanel extends React.Component {
+  static propTypes = {
+    categoryName: PropTypes.string.isRequired,
+    pois: PropTypes.array.isRequired,
+    dataSource: PropTypes.string.isRequired,
+    hasError: PropTypes.bool,
+    zoomIn: PropTypes.bool,
+    close: PropTypes.func.isRequired,
+  }
+
+  componentDidUpdate() {
+    const panelContent = document.querySelector('.panel-content');
+    if (panelContent) {
+      panelContent.scrollTop = 0;
+    }
+  }
+
+  selectPoi = poi => {
+    const previousMarker = document.querySelector('.mapboxgl-marker.active');
```

Comment:
1- n'est-il pas un peu sale de mélanger les signatures func(arg){...}   et   func = (arg) => {...} ?
2- componentDidUpdate est-ce une fonction de React appelée automatiquement, ou un truc qu'on déclare et appelle nous-même?

Reply:
1 - 
---
Il y a un sens à écrire certaines fonctions sous la forme `funct = arg => {…}`, c'est qu'ainsi elles sont autobindées à l'instance de la classe, ce qui est indispensable pour que `this` soit le bon objet quand on passe ces fonctions en callback.

Une des alternatives si on veut garder la forme `funct(arg) {…}` est de faire le binding explicitement à la main, c'est-à-dire d'écrire ce genre de choses dans le constructeur : 
```
this.selectPoi = this.selectPoi.bind(this);
this.highlightPoiMarker = this.highlightPoiMarker.bind(this);
this.onShowPhoneNumber = this.onShowPhoneNumber.bind(this);
```
Ce qui est vite assez pénible. Du coup on déclare ces fonctions en tant que champs de propriété de l'objet, grâce à cette syntaxe (qui pour ajouter au bordel est encore à l'état de proposition dans JS et requiert une [transformation Babel](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties))

Oui je sais tout ça est vraiment très tordu :/ C'est même pas lié à React mais au modèle d'objet dynamique de JS et à la façon dont il traite le contexte d'exécution des fonctions, qui peut changer.

Je te conseille la lecture de cet article pour comprendre pourquoi on en est arrivé à cette solution : https://medium.com/komenco/react-autobinding-2261a1092849
Et on peut en rediscuter si c'est pas clair :)

2 -
---

Pour `componentDidUpdate` en effet c'est une des quelques fonctions du cycle de vie d'un componsant React, appelée automatiquement après chaque `render`. C'est utile 

Voici la doc : https://reactjs.org/docs/react-component.html#componentdidupdate